walk-in-interview/src/main/java/com/google/job/data/Job.java

```diff
@@ -1,18 +1,99 @@
 package com.google.job.data;
 
+import jdk.internal.jline.internal.Nullable;
+
+import java.util.List;
+import java.util.Optional;
+
 /** Class for a job post. */
 public final class Job {
-    // TODO(issue/9): Add the feature to create a new job post
+    private static final String DUMMY = "dummy";
+
+    private String jobId;
+    // private long businessAccountId; // account is not involved so far
+    private JobStatus jobStatus;
     private String jobTitle;
+    private Location location;
+    private String jobDescription;
+    private JobPayment jobPay;
+    private List<String> requirements;
+    private long postExpiry; // a timestamp
+    private Duration jobDuration;
 
-    // No-argument constructor is needed to deserialize object when interacting with cloud firestore.
-    public Job() {}
+    private Job(String jobId, JobStatus jobStatus, String jobTitle,
+                Location location, String jobDescription,
+                JobPayment jobPayment, List<String> requirements,
+                long postExpiry, @Nullable Optional<Duration> jobDuration) {
```

Comment:
Either optional or nullable, not both. Optional is preferred, but need to check which one is accepted by firestore. Anyway, why job duration is not compulsory?

Reply:
Since we think it is ok not to specify how long the job will take